File: python/0747.至少是其他数字两倍的最大数.py
class Solution:
    # # 1. O(n) t:24ms(97%) m:14.9M(84%) 循环
    # def dominantIndex(self, nums: List[int]) -> int:
    #     f, s, idx = -1, 0, 0
    #     for i, num in enumerate(nums):
    #         if num > f:
    #             s, f = f, num
    #             idx = i
    #         elif num > s:
    #             s = num
    #     if f >= 2*s:
    #         return idx
    #     else:
    #         return -1
        
    # 2. t:32ms(65%) m:15.1M(15%) list方法
    def dominantIndex(self, nums: List[int]) -> int:
        if len(nums) <= 1: return 0
        m = max(nums)
        idx = nums.index(m)
        nums.pop(idx)
        s = max(nums)
        return idx if m >= 2*s else -1

File: java/0747.至少是其他数字两倍的最大数.java
class Solution {
    // 1. O(n) t:0ms(100%) m:36M(39%)
    public int dominantIndex(int[] nums) {
        int n = nums.length;
        if (n == 1) return 0;
        int m = -1, s = 0, idx = 0;
        for (int i = 0; i < n; i++){
            if (nums[i] > m){
                s = m;
                m = nums[i];
                idx = i;
            }else if(nums[i] > s){
                s = nums[i];
            }
        }
        return m >= 2 * s ? idx : -1;
    }
}